### ISSUE TITLE
@json Blade directive breaks with array having 2 or more keys

### ISSUE THREAD

**Issue by adam1010** [laravel/framework#25212]
Example:
```
<script type="text/javascript">
var data = '@json(["show"=>true, "revertOnUpdate" => true])';
</script>
```

Compiles to:
```
var data = '<?php echo json_encode(["show"=>true, "revertOnUpdate" => true], 512) ?>';
```

What it SHOULD compile to:
```
var data = '<?php echo json_encode(["show"=>true, "revertOnUpdate" => true], 15, 512) ?>';
```

It's supposed to be passing 512 as the 3rd argument into `json_encode` instead of the 2nd argument which acts to disable several defaults such as JSON_HEX_APOS and JSON_HEX_QUOT.

When going beyond a single key in your JSON/array, the directive will pick up the commas within the array declaration and interpret them as arguments for `json_encode`  due to its use of `explode()`

(Note: If you test with 3 keys instead of 2, it will produce a valid json_encode call, but will be missing the 2nd and 3rd parameters)

Problematic code: 
https://github.com/laravel/framework/blob/5.6/src/Illuminate/View/Compilers/Concerns/CompilesJson.php

**Comment by staudenmeir** [comment]
This is a known limitation: #24734

**Comment by adam1010** [comment]
Why don't you add the 5 lines of code to remove this "limitation"?  I'm happy to write the code to fix this bug.  All you have to do is parse out the content from within the square brackets before you explode the parameter list.

**Comment by mfn** [comment]
> Why don't you add…

Thanks @adam1010 for your suggestion, but I think for an Open Source project this isn't a nice attitude and I think…

> …I'm happy to write the code

… that this would be very much appreciated!

**Comment by adam1010** [comment]
@mfn My ticket was closed after 8hr without even any discussion of how easy it might be fix/correct. Unfortunately this has become a common occurrence for me when trying to help improve Laravel.  I spend time crafting a descriptive ticket that includes debugging information, only to be told to "go away".  I hope the maintainers can become more understanding, or devs like myself (that are actually out in the field using Laravel in production), will just work around problems locally and not bother reporting them.

**Comment by akash-mitra** [comment]
> Why don't you add the 5 lines of code to remove this "limitation"? I'm happy to write the code to fix this bug. All you have to do is parse out the content from within the square brackets before you explode the parameter list.

Couldn't agree more

**Comment by adam1010** [comment]
Forgot this was still broken in Laravel! I hacked this is my projects by overriding the Blade directive:

```
Blade::directive('json', function($expression){
      return "<?php echo json_encode($expression, JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT) ?>";
    });
```

I guess if we hurry we can get it fixed before Laravel 5.8 --- my track record of getting PRs merged isn't great so I've stopped bothering. Maybe @taylorotwell will take some interest here...
